templates/api/json.adapter.h.tpl

```diff
@@ -6,13 +6,50 @@
 #endif
 #include "api.h"
 #include "apigear/utilities/qt_native.json.adapter.h"
+
 #include <nlohmann/json.hpp>
 #include <QtCore>
 
 {{- range .Module.Imports }}
 #include "{{snake .Name}}/api/json.adapter.h"
 {{- end }}
 
+
+{{- if .Module.Externs }}
+namespace nlohmann {
+
+{{- range .Module.Externs }}
+{{- $externQt := qtExtern . }}
+
+{{- $type := $externQt.Name }}
+{{- if (not (eq $externQt.NameSpace "")) }}
+{{- $type = printf "%s::%s" $externQt.NameSpace $externQt.Name }}
+{{- end }}
+
+template<>
+struct adl_serializer<{{$type}}> {
+    static void to_json(nlohmann::json& j, const {{$type}}& p)
+	{
+		// Do serialization here
+		j = nlohmann::json{
+			//{"member_name", p.member }, ...
+		};
+    }
+
+    static void from_json(const nlohmann::json& j, {{$type}}& p)
+    {
+	    // Do deserialization here, e.g.
+	    // p.xyz = j.at("xyz").get<Int>();
+	    p = {{$type}}();
+    }
+};
+
+{{- end }}
+
+} //namespace nlohmann
+{{- end}}
+
+
```

Comment:
also one new line change too much, which just creates a lot of unnecessary changes in the generated code.